modules/dxp/apps/scim/scim-configuration-web/src/main/java/com/liferay/scim/configuration/web/internal/portlet/action/SaveScimConfigurationMVCActionCommand.java

```diff
@@ -155,6 +169,48 @@ else if (Objects.equals(cmd, "reset")) {
 
 			_oAuth2ApplicationLocalService.deleteOAuth2Application(
 				oAuth2Application);
+
+			Indexer<User> userIndexer = IndexerRegistryUtil.nullSafeGetIndexer(
+				User.class);
+
+			Indexer<UserGroup> userGroupIndexer =
+				IndexerRegistryUtil.nullSafeGetIndexer(UserGroup.class);
+
+			ActionableDynamicQuery query =
+				_expandoValueLocalService.getActionableDynamicQuery();
+
+			query.setAddCriteriaMethod(
+				dynamicQuery -> {
+					Property dataProperty = PropertyFactoryUtil.forName("data");
+
+					dynamicQuery.add(dataProperty.eq(scimClientId));
+				});
+
+			query.setPerformActionMethod(
+				(ExpandoValue expandoValue) -> {
+					_expandoRowLocalService.deleteRow(
+						expandoValue.getTableId(), expandoValue.getClassPK());
+
+					ClassName className = _classNameLocalService.getClassName(
+						expandoValue.getClassNameId());
+
+					String classNameString = className.getClassName();
```

Comment:
We can use Portal.getClassName() instead, see https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/util/PortalImpl.java#L1722-L1734